x-pack/solutions/security/plugins/security_solution/server/endpoint/services/actions/clients/sentinelone/sentinel_one_actions_client.ts

```diff
@@ -255,22 +255,12 @@ export class SentinelOneActionsClient extends ResponseActionsClientImpl {
       return cachedEntry;
     }
 
-    let s1ApiResponse: SentinelOneGetAgentsResponse | undefined;
-
-    try {
-      const response = await this.sendAction<
-        SentinelOneGetAgentsResponse,
-        SentinelOneGetAgentsParams
-      >(SUB_ACTION.GET_AGENTS, { ids: agentId });
-
-      s1ApiResponse = response.data;
-    } catch (err) {
-      throw new ResponseActionsClientError(
-        `Error while attempting to retrieve SentinelOne host with agent id [${agentId}]: ${err.message}`,
-        500,
-        err
-      );
-    }
+    const s1ApiResponse = (
+      await this.sendAction<SentinelOneGetAgentsResponse, SentinelOneGetAgentsParams>(
+        SUB_ACTION.GET_AGENTS,
+        { ids: agentId }
+      )
+    ).data;
 
```

Comment:
Could you please double-check if the error thrown by sendAction is properly handled? From reviewing the code, it’s not immediately clear.

Reply:
Yeah, I did. The `sendAction()` will already handles errors correctly and (more important) returns a `ResponseActionsClientError` that has the desired HTTP status code (`400`)